browser/mapping.go

```diff
@@ -214,6 +239,11 @@ func mapElementHandle(ctx context.Context, vu k6modules.VU, eh api.ElementHandle
 		return rt.ToValue(mehs).ToObject(rt)
 	}
 
+	jsHandleMap := mapJSHandle(ctx, vu, eh)
```

Comment:
Why do we implement `JSHandle` in `ElementHandle`? Playwright doesn't do this -- https://playwright.dev/docs/api/class-elementhandle

Reply:
If remove mapping of the `JSHandle` from the `ElementHandle`, our API would change. I want to keep the current code intact (the following). We can deal with this later if it becomes necessary.

https://github.com/grafana/xk6-browser/blob/e433cb765803dffcb77f1032ac7bd369c9e415da/api/element_handle.go#L6-L9

Also, `mapJSHandle` maps the `JSHandle` part of the `ElementHandle`. So we don't need to do it twice.

Reply:
Hmm, my preference would be to break the API so that we match Playwright. I had a look at the comments from past commits and I can't see any reason as to why we would need it. These APIs aren't documented anywhere, and only users who checked the source code would likely find them. The worst case scenario is that some users would complain that the API has broken their tests, but we can at that point put it back in -- this would prove that it is in fact useful.

I feel that this is the best time to remove this, since we're going to only add more xk6-browser users after the merge, which will make breaking things a little more challenging.

WDYT?

Reply:
I'm on the side of not complicating things right now. It's not a high priority—but the mapping is. We shouldn't block a thing that is not related to this PR and mapping in general.

We preserve the current behavior. If you want to change the API, be my guest: I've created a new issue: #728.

Reply:
Ok, np 🙂